guardrails/datatypes.py

```diff
@@ -169,7 +168,7 @@ def decorator(cls: T) -> T:
 def deprecate_type(cls: type):
```

Comment:
This can be removed since it is no longer used after the deprecated types are removed.

Reply:
The same is true for `deprecated_string_types`, `update_deprecated_type_to_string`, and the usage of `update_deprecated_type_to_string` in `List.set_children_from_xml`.